src/audio/tdfb/tdfb_direction.c

```diff
@@ -125,10 +125,8 @@ static int16_t max_mic_distance(struct tdfb_comp_data *cd)
 
 	/* Max lag based on largest array dimension. Microphone coordinates are Q4.12 meters */
 	for (i = 0; i < cd->config->num_mic_locations; i++) {
-		for (j = 0; i < cd->config->num_mic_locations; i++) {
-			if (j == i)
-				continue;
-
+		/* Start from i+1 halves the amount of iteration & to eliminate redundant checks */
```

Comment:
this comment doesn't help. Halves iterations - compared to what? The reader will not know what it was like before, they will only see the present state. So if you want any comment here you can just say, that you calculate distances between all possible microphone pairs.
In fact even in the commit description I don't think, that saying just "halves iterations" is enough, you need to explain why this works, why the other half isn't needed, i.e. that a distance between `i`th and `j`th microphones has to be calculated only once.